grafana/resource_alerting_mute_timing.go

```diff
@@ -0,0 +1,307 @@
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							DiffSuppressFunc: suppressMonthDiff,
+						},
+						"years": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: `A positive inclusive range of years, e.g. "2030" or "2025:2026".`,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func readMuteTiming(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*client).gapi
+
+	name := data.Id()
+	mt, err := client.MuteTiming(name)
+	if err != nil {
+		if strings.HasPrefix(err.Error(), "status: 404") {
+			log.Printf("[WARN] removing mute timing %s from state because it no longer exists in grafana", name)
+			data.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	data.SetId(mt.Name)
+	data.Set("name", mt.Name)
+	data.Set("intervals", packIntervals(mt.TimeIntervals))
+	return nil
+}
+
+func createMuteTiming(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*client).gapi
+
+	mt := unpackMuteTiming(data)
+	if err := client.NewMuteTiming(&mt); err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.SetId(mt.Name)
+	return readMuteTiming(ctx, data, meta)
+}
+
+func updateMuteTiming(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*client).gapi
+
+	mt := unpackMuteTiming(data)
+
+	if err := client.UpdateMuteTiming(&mt); err != nil {
+		return diag.FromErr(err)
+	}
+	return readMuteTiming(ctx, data, meta)
+}
+
+func deleteMuteTiming(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*client).gapi
+	name := data.Id()
+
+	if err := client.DeleteMuteTiming(name); err != nil {
+		return diag.FromErr(err)
+	}
+	return diag.Diagnostics{}
+}
+
+func suppressMonthDiff(k, oldValue, newValue string, d *schema.ResourceData) bool {
+	monthNums := map[string]int{
+		"january":   1,
+		"february":  2,
+		"march":     3,
+		"april":     4,
+		"may":       5,
+		"june":      6,
+		"july":      7,
+		"august":    8,
+		"september": 9,
+		"october":   10,
+		"november":  11,
+		"december":  12,
+	}
+
+	oldNormalized := oldValue
+	newNormalized := newValue
+	for k, v := range monthNums {
+		oldNormalized = strings.ReplaceAll(oldNormalized, k, fmt.Sprint(v))
+		newNormalized = strings.ReplaceAll(newNormalized, k, fmt.Sprint(v))
+	}
+
+	return oldNormalized == newNormalized
+}
+
+func unpackMuteTiming(d *schema.ResourceData) gapi.MuteTiming {
+	intervals := d.Get("intervals").([]interface{})
+	mt := gapi.MuteTiming{
+		Name:          d.Get("name").(string),
+		TimeIntervals: unpackIntervals(intervals),
+	}
+	return mt
+}
+
+func packIntervals(nts []gapi.TimeInterval) []interface{} {
+	if nts == nil {
+		return nil
+	}
+
+	intervals := make([]interface{}, 0)
```

Comment:
```suggestion
	intervals := make([]interface{}, len(nts))
```

Reply:
Doesn't quite work because we use `append`. The array ends up 2x as long as expected with the first half all empty structs. Instead I did `make([]interface{}, 0, len(nts))` which has the desired effect, size is 0 but len(nts) element preallocated.